ClientApp/src/app/study/pre-study/pre-study.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Course } from 'src/app/models/course.model';
+import { StudyOptions } from 'src/app/models/study-options.model';
+import { Word } from 'src/app/models/word.model';
+import { SnackBarService } from 'src/app/services/snack-bar.service';
+
+@Component({
+  selector: 'app-pre-study',
+  templateUrl: './pre-study.component.html',
+  styleUrls: ['./pre-study.component.css']
+})
+export class PreStudyComponent implements OnInit {
+  @Input() public studyOptions: StudyOptions;
+  @Input() public allCourses: Course[];
+  @Output() public updateSize = new EventEmitter<number>();
+  @Output() public updateType = new EventEmitter<string>();
+  @Output() public updateCourse = new EventEmitter<string>();
+  public selectedWords: Word[];
+
+  constructor(private snackBarService: SnackBarService) { }
+
+  public ngOnInit(): void {
+    this.selectedWords = this.studyOptions.course.words.slice(0, this.studyOptions.sessionSize);
+  }
+
+  public updateOptions(updatedOption: string, updatedValue: any) {
+    const invalidValueErorr = new Error('Invalid value provided.');
+
+    try {
```

Comment:
It would be cleaner to have three separate methods to handle the updating of each thing. f.ex, `updateSize` `updateCourse` `updateType`

Reply:
don't forget to add method comments